Bluetooth: shell: A2DP: fix compiling issues

gzh-terry opened the pull request:
Bluetooth: fix compiling issue when either A2DP SRC or SNK are not en…

Some of the A2DP interfaces are not defined if the corresponding feature is not enabled, and therefore shall not be initialized. This patchs include the interfaces/variables by the configurations to go through the compiling stage.

Fixes #75572

jhedberg commented:
@gzh-terry thanks for the pull request! You'll need to fix the Compliance Check issues related to your commit message before this can be merged. In particular you'll need to fix your git author information to follow the format "Firstname Lastname <email>". Once you've fixed this in your `.gitconfig` you should be able to do `git commit --amend --reset-author`. You'll also need to shorten the commit message lines according to what you see in in the Compliance check logs.

gzh-terry committed:
Bluetooth: fix compiling issue when either A2DP SRC or SNK are not enabled

Some of the A2DP interfaces are not defined if the corresponding feature
is not enabled, and therefore shall not be initialized.
This patchs include the interfaces/variables by the configurations to
go through the compiling stage.

Signed-off-by: Zihao Gao <gaozihao@xiaomi.com>

gzh-terry commented:
> @gzh-terry thanks for the pull request! You'll need to fix the Compliance Check issues related to your commit message before this can be merged. In particular you'll need to fix your git author information to follow the format "Firstname Lastname ". Once you've fixed this in your `.gitconfig` you should be able to do `git commit --amend --reset-author`. You'll also need to shorten the commit message lines according to what you see in in the Compliance check logs.

@jhedbergThanks for your prompt reply. I fixed the name and commit message issue.

gzh-terry commented:
Hi @lylezhu2012 , are all progress done for this commit? I saw the checks are passed, is there any thing to do at this stage?
This is my first time adding codes here so I'm not very familier with the workflow.
Thanks in advance.

Best regards,
Zihao

lylezhu2012 commented:
> Hi @lylezhu2012 , are all progress done for this commit? I saw the checks are passed, is there any thing to do at this stage? This is my first time adding codes here so I'm not very familier with the workflow. Thanks in advance.
> 
> Best regards, Zihao

I didn't find anything else that needed to be done.

gzh-terry commented:
> > Hi @lylezhu2012 , are all progress done for this commit? I saw the checks are passed, is there any thing to do at this stage? This is my first time adding codes here so I'm not very familier with the workflow. Thanks in advance.
> > Best regards, Zihao
> 
> I didn't find anything else that needed to be done.

Hi @lylezhu2012 ,

So I just wait someone to merge this commit?
How long does it generally takes?

Thanks,
Zihao

Thalley reviewed (approved):
Ideally the commit title would be improved, but it's not that big of a deal.

Thalley reviewed (commented):
@lylezhu2012 Classic should consider adding build tests for the shell into https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/bluetooth/shell/testcase.yaml or something to help prevent such issues in the future

jori-nordic commented:
@lylezhu2012 To merge this we still need a proper bug report, as the release is still ongoing.
When you have opened that bug report, add a "Fixes #bug-report-number" to the main text of this PR.

lylezhu2012 commented:
> @lylezhu2012 To merge this we still need a proper bug report, as the release is still ongoing. When you have opened that bug report, add a "Fixes #bug-report-number" to the main text of this PR.

Sure.

aescolar commented:
> Sure. Whether "Fixes #bug-report-number" needs to be added into commit of this PR?

It's enough to add it to the PR description
